File: yunshin/4주차/BruteForce/Programmers/84512_모음사전.js
var dictionary = []
var cadidate = ['A', 'E', 'I', 'O', 'U']

function dfs(str) {
  if (str.length > 5) return
  else if (str.length != 0) {
    dictionary.push(str)
  }

  for (char of cadidate) {
    dfs(str + char)
  }
}

function solution(word) {
  dfs('')
  dictionary.sort((a, b) => {
    for (i = 0; i < 5; i++) {
      if (a[i] != b[i]) return a[i] - b[i]
    }
  })
  return dictionary.indexOf(word) + 1
}

File: yunshin/4주차/BruteForce/Programmers/84512_모음사전.py
def solution(word):
    # global candidate,dictionary
    candidate = ['A','E','I','O','U']
    dictionary = []
    def dfs(subStr):
        if len(subStr) > 5:
            return
        else:
            if len(subStr) != 0:
                dictionary.append(subStr) 
        for can in candidate:
            dfs(subStr+can)
    dfs("")
    def customSort(x):
        for i in range(5):
            return x[i]
    dictionary.sort(key = customSort)
    answer = 0
    for el in dictionary:
        if el == word:
            return answer+1
        answer+=1